sdk/go/pulumi-language-go/language_test.go

```diff
@@ -0,0 +1,269 @@
+
+	conn, err := grpc.Dial(
+		address,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithUnaryInterceptor(rpcutil.OpenTracingClientInterceptor()),
+		grpc.WithStreamInterceptor(rpcutil.OpenTracingStreamClientInterceptor()),
+		rpcutil.GrpcChannelOptions(),
+	)
+	require.NoError(t, err)
+
+	client := testingrpc.NewLanguageTestClient(conn)
+
+	t.Cleanup(func() {
+		assert.NoError(t, cmd.Process.Kill())
+		wg.Wait()
+		// We expect this to error because we just killed it.
+		contract.IgnoreError(cmd.Wait())
+	})
+
+	engineAddress := runEngine(t)
+	return engineAddress, client
+}
+
+var expectedFailures = map[string]string{
+	"l1-empty":                           "dependency pulumi has unexpected version",
+	"l1-main":                            "error in compiling Go",
+	"l1-output-array":                    "error in compiling Go",
+	"l1-output-bool":                     "error in compiling Go",
+	"l1-output-string":                   "error in compiling Go",
+	"l1-output-number":                   "error in compiling Go",
+	"l1-stack-reference":                 "error in compiling Go",
+	"l2-target-up-with-new-dependency":   "missing go.mod",
+	"l2-destroy":                         "missing go.mod",
+	"l2-large-string":                    "missing go.mod",
+	"l2-explicit-provider":               "missing go.mod",
+	"l2-resource-alpha":                  "missing go.mod",
+	"l2-resource-simple":                 "missing go.mod",
+	"l2-resource-asset-archive":          "missing go.mod",
+	"l2-failed-create-continue-on-error": "missing go.mod",
+	"l2-engine-update-options":           "missing go.mod",
+}
+
+func TestLanguage(t *testing.T) {
+	t.Parallel()
+
+	engineAddress, engine := runTestingHost(t)
+
+	tests, err := engine.GetLanguageTests(context.Background(), &testingrpc.GetLanguageTestsRequest{})
+	require.NoError(t, err)
+
+	cancel := make(chan bool)
+
+	// Run the language plugin
+	handle, err := rpcutil.ServeWithOptions(rpcutil.ServeOptions{
+		Init: func(srv *grpc.Server) error {
+			host := newLanguageHost(engineAddress, "", "")
+			pulumirpc.RegisterLanguageRuntimeServer(srv, host)
+			return nil
+		},
+		Cancel: cancel,
+	})
+	require.NoError(t, err)
+
+	// Create a temp project dir for the test to run in
+	rootDir := t.TempDir()
+
+	snapshotDir := "./testdata/"
+
+	// Prepare to run the tests
+	prepare, err := engine.PrepareLanguageTests(context.Background(), &testingrpc.PrepareLanguageTestsRequest{
+		LanguagePluginName:   "go",
+		LanguagePluginTarget: fmt.Sprintf("127.0.0.1:%d", handle.Port),
+		TemporaryDirectory:   rootDir,
+		SnapshotDirectory:    snapshotDir,
+		CoreSdkDirectory:     "../..",
+		CoreSdkVersion:       sdk.Version.String(),
+		SnapshotEdits: []*testingrpc.PrepareLanguageTestsRequest_Replacement{ /*
+				{
+					Path:        "go\\.mod",
+					Pattern:     fmt.Sprintf("pulumi-pulumi-%s\\.tgz", sdk.Version.String()),
+					Replacement: "pulumi-pulumi-CORE.VERSION.tgz",
+				},*/
```

Comment:
```suggestion
		SnapshotEdits: []*testingrpc.PrepareLanguageTestsRequest_Replacement{
```

? Or is this still useful for somthing?

Reply:
Probably not I'll clean it up